src/main/java/uk/gov/hmcts/ccd/domain/service/common/AccessControlService.java

```diff
@@ -239,6 +252,46 @@ public CaseEventTrigger setReadOnlyOnCaseViewFieldsIfNoAccess(final CaseEventTri
         return caseEventTrigger;
     }
 
+    public CaseEventTrigger updateCollectionDisplayContextParameterByAccess(final CaseEventTrigger caseEventTrigger,
+                                                                            final Set<String> userRoles) {
+        caseEventTrigger.getCaseFields()
+            .forEach(caseViewField -> {
+                if (caseViewField.isCollectionFieldType()) {
+                    caseViewField.setDisplayContextParameter(getCollectionAccess(userRoles, caseViewField));
+                }
+
+                setChildrenCollectionDisplayContextParameter(caseViewField.getFieldType().getChildren(), userRoles);
+            });
+        return caseEventTrigger;
+    }
+
+    private void setChildrenCollectionDisplayContextParameter(final List<CaseField> caseFields,
+                                                              final Set<String> userRoles) {
+        caseFields.forEach(childField -> {
+            if (childField.isCollectionFieldType()) {
+                childField.setDisplayContextParameter(getCollectionAccess(userRoles, childField));
+            }
+
+            setChildrenCollectionDisplayContextParameter(childField.getFieldType().getChildren(), userRoles);
+        });
+    }
+
+    private String getCollectionAccess(Set<String> userRoles, CommonField field) {
+        List<String> collectionAccess = new ArrayList<>();
+        if (hasAccessControlList(userRoles, CAN_CREATE, field.getAccessControlLists())) {
+            collectionAccess.add("allowInsert");
+        }
+        if (hasAccessControlList(userRoles, CAN_DELETE, field.getAccessControlLists())) {
+            collectionAccess.add("allowDelete");
+        }
+        if (hasAccessControlList(userRoles, CAN_UPDATE, field.getAccessControlLists())) {
+            collectionAccess.add("allowInsert");
+            collectionAccess.add("allowDelete");
+        }
+        return DisplayContextParameterUtil.updateDisplayContextParameter(field.getDisplayContextParameter(),
+            DisplayContextParameterUtil.Parameter.COLLECTION, collectionAccess);
+    }
```

Comment:
I would move this method to DisplayContextParameterUtil so it know's about `"allowInsert"` and `"allowDelete" rather than AccessControlService